packages/node-http-handler/src/node-http2-handler.spec.ts

```diff
@@ -28,288 +29,420 @@ describe("NodeHttp2Handler", () => {
+        );
+        expect(establishedConnections).toBe(2);
+        expect(numRequests).toBe(2);
+        await rejects(
+          nodeH2Handler.handle(req, {}),
+          UNEXPECTEDLY_CLOSED_REGEX,
+          "should be rejected promptly due to goaway frame or destroyed connection"
+        );
+        expect(establishedConnections).toBe(3);
+        expect(numRequests).toBe(3);
+        mockH2Server3.close();
+      });
+    });
 
-      await nodeH2Handler.handle(new HttpRequest({ ...getMockReqOptions(), port: port2 }), {});
-      // @ts-ignore: access private property
-      expect(nodeH2Handler.connectionPool.size).toBe(2);
-      expect(
+    describe("destroy", () => {
+      it("destroys session and clears sessionCache", async () => {
+        await nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {});
+
+        const authority = `${protocol}//${hostname}:${port}`;
         // @ts-ignore: access private property
-        nodeH2Handler.connectionPool.get(`${protocol}//${hostname}:${port2}`)
-      ).toBeDefined();
+        const session: ClientHttp2Session = nodeH2Handler.sessionCache.get(authority)[0];
 
-      mockH2Server2.close();
+        // @ts-ignore: access private property
+        expect(nodeH2Handler.sessionCache.size).toBe(1);
+        expect(session.destroyed).toBe(false);
+        nodeH2Handler.destroy();
+        // @ts-ignore: access private property
+        expect(nodeH2Handler.sessionCache.size).toBe(0);
+        expect(session.destroyed).toBe(true);
+      });
     });
 
-    const UNEXPECTEDLY_CLOSED_REGEX = /closed|destroy|cancel|did not get a response/i;
-    it("handles goaway frames", async () => {
-      const port3 = port + 2;
-      const mockH2Server3 = createMockHttp2Server().listen(port3);
-      let establishedConnections = 0;
-      let numRequests = 0;
-      let shouldSendGoAway = true;
-
-      mockH2Server3.on("stream", (request: Http2Stream) => {
-        // transmit goaway frame without shutting down the connection
-        // to simulate an unlikely error mode.
-        numRequests += 1;
-        if (shouldSendGoAway) {
-          request.session.goaway(constants.NGHTTP2_PROTOCOL_ERROR);
-        }
+    describe("abortSignal", () => {
+      it("will not create session if request already aborted", async () => {
+        // @ts-ignore: access private property
+        expect(nodeH2Handler.sessionCache.size).toBe(0);
+        await expect(
+          nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {
+            abortSignal: {
+              aborted: true,
+              onabort: null,
+            },
+          })
+        ).rejects.toHaveProperty("name", "AbortError");
+        // @ts-ignore: access private property
+        expect(nodeH2Handler.sessionCache.size).toBe(0);
       });
-      mockH2Server3.on("connection", () => {
-        establishedConnections += 1;
+
+      it("will not create request on session if request already aborted", async () => {
```

Comment:
Does this make sense. why do we need to create a new session if a request is aborted?

Reply:
This test verifies that new request is not created on existing session, if request is already aborted. It does not create a new session.
* A session is created when first request is made on line 252.
* Second request is not made when handle is called on line 259.